plugins/module_utils/openshift_builds.py

```diff
@@ -0,0 +1,431 @@
+                msg = "Build %s/%s has not complete after %d second(s)," \
+                      "current status is %s" % (namespace, name, wait_timeout, last_status_phase)
+
+                self.fail_json(msg=msg)
+
+        result = [result] if result else []
+        self.exit_json(changed=True, builds=result)
+
+    def cancel_build(self, restart):
+
+        kind = 'Build'
+        api_version = 'build.openshift.io/v1'
+
+        namespace = self.params.get("namespace")
+        phases = []
+        possible_phases = ('new', 'pending', 'running')
+
+        for v in self.params.get("build_phases"):
+            if v.lower() not in possible_phases:
+                self.fail_json(msg="Invalid build_phase value '%s', must be one of 'new', 'pending', or 'running'." % v)
+            phases.append(v.lower())
+        phases = phases or possible_phases
+
+        names = []
+        if self.params.get("build_name"):
+            names.append(self.params.get("build_name"))
+        else:
+            build_config = self.params.get("build_config_name")
+            # list all builds from namespace
+            params = dict(
+                kind=kind,
+                api_version=api_version,
+                namespace=namespace
+            )
+            resources = self.kubernetes_facts(**params).get("resources", [])
+
+            def _filter_builds(build):
+                config = build["metadata"].get("labels", {}).get("openshift.io/build-config.name")
+                return config in build_config
+
+            for item in list(filter(_filter_builds, resources)):
+                name = item["metadata"]["name"]
+                if name not in names:
+                    names.append(name)
+
+        if len(names) == 0:
+            self.exit_json(changed=False, msg="No Build found from namespace %s" % namespace)
+
+        warning = []
+        builds_to_cancel = []
+        for name in names:
+            params = dict(
+                kind=kind,
+                api_version=api_version,
+                name=name,
+                namespace=namespace
+            )
+
+            resource = self.kubernetes_facts(**params).get("resources", [])
+            if len(resource) == 0:
+                warning.append("Build %s/%s not found" % (namespace, name))
+                continue
+
+            resource = resource[0]
+            phase = resource["status"].get("phase").lower()
+
+            # Build status.phase is matching the requested state and is not completed
+            if phase in phases:
+                builds_to_cancel.append(resource)
+            else:
+                warning.append("build %s/%s is not in expected phase, found %s" % (namespace, name, phase))
+
+        changed = False
+        result = []
+        for build in builds_to_cancel:
+            # Set cancelled to true
+            build["status"]["cancelled"] = True
+            name = build["metadata"]["name"]
+            changed = True
```

Comment:
Is check_mode handled somewhere for cancel_build() and start_build() ?

Reply:
the openshift_build module does not support check_mode for now, this can be included later